R/getCode.R

```diff
@@ -1,97 +1,110 @@
+    if (grepl(pattern, x)) {
+      extras <- c("download", "convert", "correct")
       for (e in extras) {
-        tmp <- try(deparse(eval(parse(text = sub(pattern,paste0("\\1",e),x)))),silent = TRUE)
-        if (!("try-error" %in% class(tmp))) out <- c(out,tmp)
+        tmp <- try(deparse(eval(parse(text = sub(pattern, paste0("\\1", e), x)))), silent = TRUE)
+        if (!("try-error" %in% class(tmp))) out <- c(out, tmp)
       }
     }
-    return(paste(out,collapse = " "))
+    return(paste(out, collapse = " "))
   }
-  
+
   # extract function pool
   fpool <- getCalculations("read|calc|full|tool", packages = packages, globalenv = globalenv)
-  
+
   # check for duplicates
-  fpool$fname <- sub("^.*:::","",fpool$call)
+  fpool$fname <- sub("^.*:::", "", fpool$call)
   duplicated <- duplicated(fpool$fname, fromLast = TRUE)
   if (any(duplicated)) {
-    base::warning("Duplicate entries found for ",paste(fpool$fname[duplicated],collapse = ", "),
-                  "! Last entry will be used!")
-    fpool <- fpool[!duplicated,]
+    for (d in fpool$fname[duplicated]) {
+      warning("Duplicate functions found: ", paste(fpool$call[fpool$fname == d], collapse = ", "),
+              "! Laste entry will be used")
+    }
+    fpool <- fpool[!duplicated, ]
   }
-  
+
   # read in source code
-  code <- sapply(fpool$call, .extractCode)
-  
-  
+  code <- sapply(fpool$call, .extractCode)  # nolint
+
+
   .getMappingFiles <- function(code) {
-    code <- code[!(names(code) %in% paste0("madrat:::",c("toolGetMapping", "toolConvertMapping", "toolAggregate")))]
-    getMappings <- stri_extract_all(code, regex = "toolGetMapping\\(([^()]*|[^(]*\\([^)]*\\)[^)]*)\\)", omit_no_match = TRUE)
+    code <- code[!(names(code) %in% paste0("madrat:::", c("toolGetMapping", "toolConvertMapping", "toolAggregate")))]
+    getMappings <- stri_extract_all(code, regex = "toolGetMapping\\(([^()]*|[^(]*\\([^)]*\\)[^)]*)\\)",
+                                    omit_no_match = TRUE)
     names(getMappings) <- names(code)
-    getMappings <- getMappings[sapply(getMappings, length) > 0]
+    getMappings <- getMappings[vapply(getMappings, length, integer(1)) > 0]
     .tmp <- function(x) {
       out <- try(eval(parse(text = x)), silent = TRUE)
       if ("try-error" %in% class(out)) return("NOTFOUND")
       return(out)
     }
     .evals <- Vectorize(.tmp, "x", USE.NAMES = FALSE)
     .getPaths <- function(x) {
-      x <- gsub(" +"," ",x)
-      x <- sub(", ?returnPathOnly ?= ?(FALSE|TRUE)","", x)
+      x <- gsub(" +", " ", x)
+      x <- sub(", ?returnPathOnly ?= ?(FALSE|TRUE)", "", x)
       x <- sub("\\)$", ", returnPathOnly = TRUE)", x)
       return(unique(.evals(unique(x))))
     }
-    getMappings <- lapply(getMappings,.getPaths)
+    getMappings <- lapply(getMappings, .getPaths)
+    .warnNotfound <- function(getMappings) {
+      .notfound <- function(set) return(is.element("NOTFOUND", set))
+      notfound <- vapply(getMappings, .notfound, logical(1))
```

Comment:
```suggestion
      notfound <- vapply(getMappings, function(set) is.element("NOTFOUND", set), logical(1))
```
I'd just inline the function here. In any case I'd omit the explicit `return`.

Reply:
fun fact: in R 4.1 you could even use the new short function syntax
```suggestion
      notfound <- vapply(getMappings, \(set) is.element("NOTFOUND", set), logical(1))
```

Reply:
not sure what the advantage would be